src/sentry/middleware/integrations/parsers/github.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+import logging
+
+from sentry.middleware.integrations.parsers.base import BaseRequestParser
+from sentry.models.integrations.integration import Integration
+from sentry.models.outbox import WebhookProviderIdentifier
+from sentry.services.hybrid_cloud.util import control_silo_function
+from sentry.types.integrations import EXTERNAL_PROVIDERS, ExternalProviders
+from sentry.utils import json
+
+logger = logging.getLogger(__name__)
+
+
+class GithubRequestParser(BaseRequestParser):
+    provider = EXTERNAL_PROVIDERS[ExternalProviders.GITHUB]
+    webhook_identifier = WebhookProviderIdentifier.GITHUB
+
+    @control_silo_function
+    def get_integration_from_request(self) -> Integration | None:
+        if not self.is_json_request():
+            return None
+        try:
+            event = json.loads(self.request.body.decode(encoding="utf-8"))
+        except json.JSONDecodeError:
+            return None
+        external_id = event.get("installation", {}).get("id")
+        return Integration.objects.filter(external_id=external_id, provider=self.provider).first()
+
+    def get_response(self):
+        # All github webhooks will be sent to region silos
+        regions = self.get_regions_from_organizations()
+        if len(regions) == 0:
+            logger.error("no_regions", extra={"path": self.request.path})
```

Comment:
Don't integration installation requests get handled by control silo? Will they enter this path and log an error?

Reply:
Forgot to mention, but this is addressed [here](https://github.com/getsentry/sentry/pull/49651/files#diff-2b23119e9f8cf0fede77bd00fd52dece5e9b042f771f369f5ece7d5c0a30e444) in https://github.com/getsentry/sentry/pull/49651 and changed for slack as well. The `/extensions/{provider}/setup/` path is used for installation for most integrations via `PipelineAdvancerView`, and I just made it so that those requests always get handled by the control silo, and don't need to get parsed (especially since the integration doesn't exist in the db at that stage)